src/main/scala/com/kinja/soy/SoyMacroImpl.scala

```diff
@@ -53,11 +53,23 @@ object SoyMacroImpl {
     }
 
     val applies =
-      companionType.declaration(stringToTermName("apply")) match {
+      companionType.declaration(newTermName("apply")) match {
         case NoSymbol => c.abort(c.enclosingPosition, "No apply function found")
         case s => s.asMethod.alternatives
       }
 
+    // When presented with a case class parameterized over A, the A taken by
+    // `apply` and the A returned by `unapply` do not compare equal because they
+    // have different owners. In this case we compare the types of the symbols
+    // which are both className.A and compare equal.
+    def compareTypeLists(a: List[Type], b: List[Type]): Boolean =
```

Comment:
The problem here is that the `A` in `apply`
```scala
def apply[A](i : Int, a : A) : Foo[A]
```
and the `A` in `unapply`
```scala
def unapply[A](foo : Foo[A]) : Option[(Int, A)]
```
Don't compare equal since one belongs to `apply` and the other to `unapply` but they both have the same underlying type which is `Foo.A`.

I fixed that by checking for type arguments and then comparing their underlying types instead. I'm worried that this approach does not work generally so be on the lookout for false matches.